### ISSUE TITLE
[YSQL] Fix inequality operator on range primary key

### ISSUE THREAD

**Issue by jj-kim** [yugabyte/yugabyte-db#3106]
When running some statements with a condition including an inequality operator on a range primary key column, the operator seems to act as an _equals_ operator instead.  Only some statements seem to be affected, and I check them below:

* [x] `DELETE`
* [ ] `INSERT`
* [ ] `SELECT`
* [x] `UPDATE`

Here is an example with `DELETE`:

```sql
CREATE TABLE a (i int, PRIMARY KEY (i ASC));
INSERT INTO a VALUES (1), (2), (3);
DELETE FROM a WHERE i < 2;
SELECT * FROM a;
```

```
 i
---
 1
 3
(2 rows)
```

```sql
DELETE FROM a WHERE i <= 2;
SELECT * FROM a;
```

```
 i
---
 1
 3
(2 rows)
```

Here is an example with `INSERT` `ON CONFLICT DO UPDATE`:

```sql
CREATE TABLE c (i int, j int, PRIMARY KEY (i DESC));
INSERT INTO c VALUES (1, 10), (2, 20), (3, 30);
INSERT INTO c VALUES (3, 99) ON CONFLICT (i) DO UPDATE SET j = 77 WHERE c.i < 2;
SELECT * FROM c;
```

```
 i | j
---+----
 3 | 30
 2 | 20
 1 | 10
(3 rows)
```

```sql
INSERT INTO c VALUES (3, 99) ON CONFLICT (i) DO UPDATE SET j = 77 WHERE c.i > 2;
SELECT * FROM c;
```

```
 i | j
---+----
 3 | 77
 2 | 20
 1 | 10
(3 rows)
```

Here is an example with `UPDATE`:

```sql
CREATE TABLE b (i int, j int, PRIMARY KEY (i DESC));
INSERT INTO b VALUES (1, 10), (2, 20), (3, 30);
UPDATE b SET j = 100 WHERE i < 2;
SELECT * FROM b;
```

```
 i |  j
---+-----
 3 |  30
 2 | 100
 1 |  10
(3 rows)
```

**Comment by ndeodhar** [comment]
The `INSERT ON CONFLICT DO UPDATE` examples are expected. The behavior is same in postgres too.

Update and Delete are bugs and most likely related to commit https://github.com/yugabyte/yugabyte-db/commit/564ddd1470d6e35afb37543b26a464774cafb6f5

**Comment by jj-kim** [comment]
Right, `INSERT` seems fine, just as I have not checked it in the bullet list.

**Comment by m-iancu** [comment]
Fixed by 7e222b4.
